CDAP-4959, CDAP-7758 Adjust versions used by docs and Cask Market docs

awholegunch opened the pull request:
- Updates the versions used by the release for Hydrator, Tracker, Guides et al.
  - Consolidates all the "magic" values set in various config files into the ``vars`` file
   - Makes the config files (``conf.py``) as identical as possible
  - Adds additional versioning info to log files
  - Removes usage of a "max" version of Node.js, as it is not currently known.
- Cask Market documentation
  - Adds versioning to the Cask Market documentation
  - Fixes errors
  - Consolidates the pages into a single page
- Cask Tracker documentation
  - Fixes a typo

Running as a Quick Build: http://builds.cask.co/browse/CDAP-DQB189-8 (passes)

Page of interest: Cask Market documentation: http://builds.cask.co/artifact/CDAP-DQB189/shared/build-8/Docs-HTML/4.0.0-SNAPSHOT/en/integrations/cask-market.html

Fix for:
- https://issues.cask.co/browse/CDAP-4959
- https://issues.cask.co/browse/CDAP-7245
- https://issues.cask.co/browse/CDAP-7758

Commit:
Updates the versions used by the release.
Adds versioning to the Cask Market documentation.
Fixes errors.
Consolidates the pages into a single page.

Commit:
Improve reporting of values in log file.

Commit:
Move all magic values to the vars file.
Made all "conf.py" files as identical as possible.
Fixed a typo in the Tracker documentation.

Commit:
Merge branch 'release/4.0' into docs/r4.0_7758_hard_coded_values

Commit:
Cleanup vars and use the same style as the other repos.

wolf31o2 commented on cdap-docs/vars:
```diff
@@ -26,33 +26,37 @@ BUILD_RST_HASH_LOCATION="cdap-docs/vars"
 #
 
+GIT_NODE_JS_MIN_VERSION="beginning with v0.10.36"
+GIT_NODE_JS_MAX_VERSION="v4.4.0"
```
Can we set these to match CDAP-7245

awholegunch replied:
Fixed.

Commit:
Update vars

Setting min Node.js  version.

Commit:
Update common_conf.py

Split Node.js versions (max and min) so that they can be set independently. That way, one can be set, and if the other is not set and used, Sphinx will flag it with a warning about using an unset replacement value.

sreevatsanraman commented:
LGTM

Commit:
Remove usage of max version for Node.js
